### ISSUE TITLE
The readings of artist (Directory) are not updated

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1110]
## Problem description

The readings of artist (Directory) are not updated. If the readings tag of artist name is updated repeatedly, a false search may occur. It especially affects phonological search and indexing.

### Steps to reproduce

1. Scan files with artist names and album artist readings.
2. Change the reading tag, and scan.
3. In this case, even if the search is performed, the search using the new tag cannot be performed. (Since it is possible to search by name, it may be difficult for the user to notice this defect. )

## System information

 * **Jpsonic version**: Probably the first implementation of a feature that supplements directory reading. (v107.2.0~)

## Additional notes

The implementation is complex, but the specification is less complex. 

https://tesshu.com/jpsonic/spec/scan-specifications

Currently, albums and song records are updated with tags without any problems, and the scan results follow the user's edits. Search works as expected. However, the field of "reading"  used for the Artist (Directory) search of the file structure does not have a corresponding tag in the first place. Therefore, the reading values  of artist name are estimated in the order of priority of scanned tags of file modification date, album artist, artist, and composer, and if they do not exist, the analysis value is set.

 - If the Artist (Directory) reading is undesigned, set a candidate value.
 - In other words, it works as expected only once at the beginning. :kissing_smiling_eyes:
